pkg/api/defaults-cloud-controller-manager.go

```diff
@@ -5,18 +5,27 @@ package api
 
 import (
 	"strconv"
+	"strings"
 
 	"github.com/Azure/aks-engine/pkg/api/common"
 )
 
 func (cs *ContainerService) setCloudControllerManagerConfig() {
 	o := cs.Properties.OrchestratorProfile
+	isAzureCNIDualStack := cs.Properties.IsAzureCNIDualStack()
+	clusterCidr := o.KubernetesConfig.ClusterSubnet
+	if isAzureCNIDualStack {
+		clusterSubnets := strings.Split(clusterCidr, ",")
+		if len(clusterSubnets) > 1 {
+			clusterCidr = clusterSubnets[1]
+		}
+	}
 	staticCloudControllerManagerConfig := map[string]string{
-		"--allocate-node-cidrs":         strconv.FormatBool(!o.IsAzureCNI()),
-		"--configure-cloud-routes":      strconv.FormatBool(o.RequireRouteTable()),
+		"--allocate-node-cidrs":         strconv.FormatBool(!o.IsAzureCNI() || isAzureCNIDualStack),
+		"--configure-cloud-routes":      strconv.FormatBool(o.RequireRouteTable() || isAzureCNIDualStack),
```

Comment:
Let's move this additional `isAzureCNIDualStack` criteria into the `RequireRouteTable` method itself, then we'll get that new criteria for free in all the other areas that depend upon `RequireRouteTable()`

Reply:
Yes i thought about this but there is one concern. Need to move RequireRouteTable as function of Properties instead of OrchestratorProfile . Is that fine?

func (p *Properties) IsAzureCNIDualStack() bool {
	o := p.OrchestratorProfile
	f := p.FeatureFlags
	return o.IsAzureCNI() && f.IsFeatureEnabled("EnableIPv6DualStack")
}

Reply:
Good catch. Yes, that's O.K.

Reply:
just confirming this OK is meant for moving RequireRouteTable as function of Properties?

Reply:
Yep, I am definitely encouraging you to refactor RequireRouteTable in order to support using the isAzureCNIDualStack Properties method. Hopefully it won't be too tricky to update all usages of RequireRouteTable. If so, lemme know.

Reply:
done